Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -147,45 +151,88 @@ class TouchIDSetting: Setting {
         // we can disable interaction of the switch and still handle tap events.
         let control = UISwitch()
         control.onTintColor = UIConstants.ControlTintColor
-        control.on = authInfo?.useTouchID ?? false
+        self.authInfo = KeychainWrapper.authenticationInfo()
+        control.on = authenticationDelegate?.authInfoUseAuthentication ?? false
         control.userInteractionEnabled = false
         switchControl = control
 
         let accessoryContainer = UIView(frame: control.frame)
         accessoryContainer.addSubview(control)
 
-        let gesture = UITapGestureRecognizer(target: self, action: #selector(TouchIDSetting.switchTapped))
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(AuthenticationSetting.switchTapped))
         accessoryContainer.addGestureRecognizer(gesture)
 
         cell.accessoryView = accessoryContainer
     }
 
     @objc private func switchTapped() {
+        self.authInfo = KeychainWrapper.authenticationInfo()
```

Comment:
Same comment as above. Not sure if we need to re-cache this when we tap the switch. If we need a new value at the moment of tapping probably better off to store in a local var instead. If not, you can used the cached copy.